jsk_perception/docker/ofa/server.py

```diff
@@ -0,0 +1,278 @@
+            image = cv2.resize(img, dsize=(640, 480)) # NOTE forcely
+            image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+            image = Image.fromarray(image)
+            # Construct input sample & preprocess for GPU if cuda available
+            sample = self.construct_sample(image, text)
+            sample = utils.move_to_cuda(sample) if self.use_cuda else sample
+            sample = utils.apply_to_sample(apply_half, sample) if self.use_fp16 else sample
+            if self.task_name == "caption":
+                with torch.no_grad():
+                    result, scores = eval_step(self.task, self.generator, self.models, sample)
+                text = result[0]['caption']
+                return text
+            elif self.task_name == "vqa_gen":
+                with torch.no_grad():
+                    result, scores = zero_shot_step(self.task, self.generator, self.models, sample)
+                text = result[0]['answer']
+                return text
+        elif self.task_name == "refcoco":
+            pass
+
+# run
+if __name__ == "__main__":
+    app = Flask(__name__)
+    ofa_task = os.environ["OFA_TASK"]
+    ofa_model_scale = os.environ["OFA_MODEL_SCALE"]
+    # TODO add refcoco
+    if ofa_task == "all":
+        caption_infer = Inference("caption", ofa_model_scale)
+        vqa_infer = Inference("vqa_gen", ofa_model_scale)
+
+    elif ofa_task == "caption":
+        caption_infer = Inference("caption", ofa_model_scale)
+
+    elif ofa_task == "vqa_gen":
+        vqa_infer = Inference("vqa_gen", ofa_model_scale)
+
+    else:
+        raise RuntimeError("No application is available")
+
+    try:
+        @app.route("/caption", methods=['POST'])
+        def caption_request():
+            data = request.data.decode("utf-8")
+            data_json = json.loads(data)
+            # process image
+            image_b = data_json['image']
+            image_dec = base64.b64decode(image_b)
+            data_np = np.fromstring(image_dec, dtype='uint8')
+            img = cv2.imdecode(data_np, 1)
+            # get text
+            texts = data_json['queries']
+            results = []
+            for text in texts:
+                answer = caption_infer.infer(img, text)
+                results.append({"question": text, "answer": answer})
+            return Response(response=json.dumps({"results": results}), status=200)
+    except NameError:
+        print("Skipping create caption app")
+
+    try:
+        @app.route("/vqa_gen", methods=['POST'])
+        def vqa_request():
+            data = request.data.decode("utf-8")
+            data_json = json.loads(data)
+            # process image
+            image_b = data_json['image']
+            image_dec = base64.b64decode(image_b)
+            data_np = np.fromstring(image_dec, dtype='uint8')
+            img = cv2.imdecode(data_np, 1)
+            # get text
+            texts = data_json['queries']
```

Comment:
Shouldn't this be `questions` for the VQA? https://github.com/mqcmd196/jsk_recognition/blob/PR/large-scale-vil/jsk_recognition_msgs/action/VQATask.action#L2

```
  text: "Exception in execute callback: 'VQATaskGoal' object has no attribute 'queries'"
```

@mqcmd196

Reply:
Or is the actionlib implementation deprecated and setting the dynamic reconfigure is the intended usage?